### DISCUSSION TITLE
PackageNotFound Error when using a module alias

### DISCUSSION THREAD

**Discussion by jsjoeio** [oven-sh/bun#3606] [Q&A]
We have a monorepo setup with Turborepo. We're using bun to run a TypeScript script as a cron job. 

We recently moved some code into our /shared directory and now one of the files in the project uses a module alias. When we do:
```shell
bun scripts/index.ts
```

It fails with this error:

```zig
[0.02ms] ".env"
  🔍 Resolving [1/2] 1 | import { `GithubUsername` } from '@earnifi/shared/src/classes/GithubUsername/GithubUsername';
                                   ^
PackageNotFound: Package '@earnifi/shared' was not found
 url: "https://registry.npmjs.org/@earnifi/shared"
 pkg: "@earnifi/shared"

      at /Users/jp/dev/path/to/file.ts
```

Here is what the directory structure looks like:

```tree
── apps
│   ├── express
├── packages
│   ├── shared

```
The script lives in `apps/express` and that's where i'm running the `bun` command.

**Comment by Jarred-Sumner** [comment]
How is the module alias set? Is there a package.json with a `"name"` field in `packages/shared`? Is tsconfig.json `"paths"` configured in the project root (or in `"extends"` inside `@earnifi/express`)? Does the workspace install all the packages? 

When no node_modules folder exists, Bun will attempt to automatically install the package from npm. That it went to npm for the package means it wasn't configured to alias the package

**Reply by jsjoeio** [reply]
Good questions! I'm trying to figure that out as well but here's my understanding based on what I'm seeing.

Turborepo sees it as an [Internal Package](https://turbo.build/repo/docs/handbook/sharing-code/internal-packages#our-first-internal-package). 

> Is there a package.json with a "name" field in packages/shared

Yes, inside `packages/shared` is a `package.json` with these keys:

```json
{
  "name": "@earnifi/shared",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
}
```

Then inside the `package.json` of `apps/express`, the package is listed under `"dependencies"` as ` "@earnifi/shared": "^0.0.0",`.

> Does the workspace install all the packages? 

I believe so. Typically I'll run `npm install` from the root and Turborepo will run that task in all apps and packages. 

We originally were using `bun` to speed up running this script in CI by not having to install all the dependencies first.
